connectors/spark/src/main/scala/io/unitycatalog/spark/auth/GcsVendedTokenProvider.java

```diff
@@ -0,0 +1,85 @@
+package io.unitycatalog.spark.auth;
+
+import com.google.cloud.hadoop.util.AccessTokenProvider;
+import io.unitycatalog.client.model.GcpOauthToken;
+import io.unitycatalog.spark.UCHadoopConf;
+import io.unitycatalog.spark.utils.Clock;
+import org.apache.hadoop.conf.Configuration;
+import org.sparkproject.guava.base.Preconditions;
+
+import java.io.IOException;
+import java.time.Instant;
+
+public class GcsVendedTokenProvider extends GenericCredentialProvider
+    implements AccessTokenProvider {
+  public static final String ACCESS_TOKEN_KEY = "fs.gs.auth.access.token.credential";
+  public static final String ACCESS_TOKEN_EXPIRATION_KEY = "fs.gs.auth.access.token.expiration";
+
+  private Configuration conf;
+
+  /**
+   * Constructor for the hadoop's CredentialProviderListFactory to initialize.
+   */
+  public GcsVendedTokenProvider() {
+    super();
+  }
+
+  GcsVendedTokenProvider(Clock clock, long renewalLeadTimeMillis) {
+    super(clock, renewalLeadTimeMillis);
+  }
+
+  @Override
+  public GenericCredential initGenericCredential(Configuration conf) {
+    if (conf.get(UCHadoopConf.GCS_INIT_OAUTH_TOKEN) != null
+        && conf.get(UCHadoopConf.GCS_INIT_OAUTH_TOKEN_EXPIRED_TIME) != null) {
```

Comment:
We don't need the expiration time nullable check here,  since it's allowed to be nullable if it's a token/credential without expiration.  I mean the static credentials. 

You can see this comment before: https://github.com/unitycatalog/unitycatalog/pull/1123/files#r2460798294